WPMedia/arc-themes-components
# Tmedia 676 input component

## Ticket

- [TMEDIA-676](https://arcpublishing.atlassian.net/browse/TMEDIA-676)

## Description

Add Input component

## Acceptance Criteria

Create an <Input> component

Supported types: email, password, and text

Input includes label, input area, placeholder text, help text, and value

Includes state variations of default, warning, success and error 

Have Scss file that adds the hooks to the Sass library

Have test coverage

Have storybook documentation




## Test Steps

1. Checkout branch - `git checkout TMEDIA-676-input-component`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the Input storybook documentation
5. Look at the example mocks to validate that the variations use make sense visually of what's possible. 

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
